src/lib/mongoBackend/mongoSubscribeContext.cpp

```diff
@@ -175,6 +178,23 @@ HttpStatusCode mongoSubscribeContext
 
     reqSemGive(__FUNCTION__, "ngsi10 subscribe request", reqSemTaken);
 
+    
+    //
+    // Add the subscription to the subscription cache.
+    // But only if any of the entities in entityIdVector is pattern based.
```

Comment:
Not only pattern based... you also have to check that if it is ONCHANGE.

Maybe the checking could be done in the insert() method of the cache class, so this part of the code in the mongoBackend keeps clean (just `subCache->insert(...)`).